### PR TITLE
[Merged by Bors] - TY-2743 clean sources csv

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-2743]

**Summary**

- add script to clean the sources csv file from empty names and irrelevant fields
- the cleaned sources csv file is attached to the task

[TY-2743]: https://xainag.atlassian.net/browse/TY-2743?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/tooling/bin/clean_sources.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    let mut writer = BufWriter::new(File::create(out_file)?);
+
+    let mut broken_line = String::new();
+    for line in reader.lines() {
+        let line = line?;
+        if line.split(';').count() == 4 {
+            writer.write_all(line.as_bytes())?;
+            writer.write_all(&[b'\n'])?;
+        } else {
+            broken_line.push_str(&line);
```
Comment (Robert-Steiner) [inline discovery_engine_core/tooling/bin/clean_sources.rs:49]: 🔧 It looks like that the illegal linebreaks mostly occur in the name of the record.
e.g.

```
Aktuelles Programm
- SWR Fernsehen;swrfernsehen.de;False;47911
```

The problem is that the next line contains three `;` so we only keep `- SWR Fernsehen;swrfernsehen.de` and `broken_line` will just append the lines that contain the linebreak.

**Review by Robert-Steiner — COMMENTED** [review]
looks good! there is only one small issue with the linebreaks

File: discovery_engine_core/tooling/bin/clean_sources.rs (hunk 1 above)
Comment (janpetschexain) [reply discovery_engine_core/tooling/bin/clean_sources.rs:49]: good catch 👍 i'll change it such that we merge consecutive lines if there have been broken lines before, this should fit the current issue.

File: discovery_engine_core/tooling/bin/clean_sources.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                broken_line = line;
+            } else {
+                broken_line.push(' ');
+                broken_line.push_str(&line);
+            }
+            if broken_line.split(';').count() == 4 {
+                writer.write_all(broken_line.as_bytes())?;
+                writer.write_all(&[b'\n'])?;
+                broken_line.clear();
+            }
```
Comment (Robert-Steiner) [inline discovery_engine_core/tooling/bin/clean_sources.rs:57]: ❔I'm not sure if it makes sense to return an error/panic, if we for some reason get `broken_line.split(';').count() > 4` wdyt?

**Review by Robert-Steiner — APPROVED** [review]

File: discovery_engine_core/tooling/bin/clean_sources.rs (hunk 2 above)
Comment (janpetschexain) [reply discovery_engine_core/tooling/bin/clean_sources.rs:57]: in general we should get wellformed csv files, even the broken lines shouldn't be there, but unfortunately it's not up to us to ensure this. for now i would just keep the script as is, if we need to we can always update it later on.

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**